### PR TITLE
[Workflow] config doc

### PR DESCRIPTION
**Pull request by noniagriconomie**
Hi,

This draft PR follows this one
https://github.com/symfony/symfony-docs/pull/11492/files (config for 3.X version)
but for new version, mostly 4.3

I checked this PR also
https://github.com/symfony/symfony/pull/30890, @lyrixx review appreciated please

Maybe to merge after merging https://github.com/symfony/symfony-docs/pull/11578, and with a rebase ?

--------------------

Draft for now

- Closes https://github.com/symfony/symfony-docs/issues/11229

this one can be closed also:

- Closes https://github.com/symfony/symfony-docs/issues/9475
- Closes https://github.com/symfony/symfony-docs/issues/10197

### CODE REVIEW

**Comment by OskarStark** [comment]
#11578 is merged, you can now rebase your branch 😃

**Review by lyrixx — APPROVED** [review]

File: workflow.rst [hunk 1]
```diff
@@ -134,8 +134,8 @@ like this:
                         'enabled' => true
                     ],
                     'marking_store' => [
-                        'type' => 'multiple_state', // or 'single_state'
```
Comment (OskarStark) [inline workflow.rst]: @noniagriconomie I am wondering if there is no need for changing the XML config 🤔 in this example...

Comment (noniagriconomie) [reply workflow.rst]: @OskarStark what dou you mean? i fear not understanding you

Comment (OskarStark) [reply workflow.rst]: I meant, you update the YAML, and PHP example, but not the XML config in between...

Comment (noniagriconomie) [reply workflow.rst]: understood, will definitly patch it

Comment (OskarStark) [reply workflow.rst]: Thank you 👍🏻

File: workflow.rst [hunk 2]
```diff
@@ -90,35 +90,28 @@ like this:
...
                 <framework:workflow name="blog_publishing" type="workflow">
                     <framework:audit-trail enabled="true"/>
-
                     <framework:marking-store type="single_state">
                         <framework:argument>currentPlace</framework:argument>
                     </framework:marking-store>
-
                     <framework:support>App\Entity\BlogPost</framework:support>
-
+                    <framework:initial_marking>draft</framework:initial_marking>
```
Comment (noniagriconomie) [inline workflow.rst]: @OskarStark really unsure of the pattern, I never use xml :/
Also i remove spaces like php and yml docbloc

File: workflow.rst [hunk 3]
```diff
@@ -630,8 +612,8 @@ requires:
                         <framework:from>reviewed</framework:from>
                         <framework:to>published</framework:to>
                         <framework:metadata>
-                            <framework:hour_limit>20</framework:priority>
-                            <framework:explanation>You can not publish after 8 PM.</framework:priority>
+                            <framework:hour_limit>20</framework:hour_limit>
```
Comment (noniagriconomie) [inline workflow.rst]: Also, I check this i made mistake documenting metadata for xml, see here
https://github.com/symfony/symfony-docs/pull/11580/files#diff-c3ca19ec1845c786182534bf1afe2414R615
this change should go for 4.2 as well

File: workflow/introduction.rst [hunk 4]
```diff
@@ -190,46 +190,46 @@ Below is the configuration for the pull request state machine.
...
-                    'test',
-                    'review',
-                    'merged',
-                    'closed',
-                  ],
-                  'transitions' => [
-                    'submit'=> [
-                      'from' => 'start',
-                      'to' => 'test',
+                    'type' => 'state_machine',
```
Comment (noniagriconomie) [inline workflow/introduction.rst]: this change is **only** a clean reindent

**Comment by noniagriconomie** [comment]
@OskarStark friendly ping, do i need to do other thing? thx

File: workflow.rst (hunk 2 above)
Comment (OskarStark) [reply workflow.rst]: I am unsure too about the xml config

cc @lyrixx

**Comment by lyrixx** [comment]
Hello @noniagriconomie 
Why did you close your PR?

**Comment by noniagriconomie** [comment]
@lyrixx fail branch removal, thank for noticing :)

**Comment by OskarStark** [comment]
Argh wrong button

File: workflow.rst (hunk 3 above)
Comment (OskarStark) [reply workflow.rst]: As `4.2` is end of maintenance now, lets ignore it for this branch!

File: workflow.rst (hunk 2 above)
Comment (OskarStark) [reply workflow.rst]: friendly ping @lyrixx can you please give us some feedback here?

**Comment by OskarStark** [comment]
@noniagriconomie can you please rebase your branch? Afterwards we are good to go 👍

**Comment by noniagriconomie** [comment]
@OskarStark will do it asap, but i fear i will be afk soon ofr a short period of time :/
will do my best :)

**Comment by noniagriconomie** [comment]
@OskarStark PR is ok for me

File: reference/configuration/framework.rst [hunk 5]
```diff
@@ -255,8 +255,9 @@ Configuration
   * :ref:`name <reference-workflows-name>`
 
     * `audit_trail`_
-    * `initial_place`_
```
Comment (OskarStark) [inline reference/configuration/framework.rst]: @javiereguiluz is it ok to remove the deprecated one here? Or do we need to leave some info about the before option name?

**Comment by wouterj** [comment]
Soo as you can see I completely destroyed your branch here. Fortunately, GitHub keeps all commit refs in your fork, so I was able to correctly rescue all your amazing changes and created a new PR: https://github.com/symfony/symfony-docs/pull/12146

I've also checked the XML code and it looks perfect. One minor thing: While both underscore and dash are valid in Symfony configuration, using the dash is considered better XML, so I've directly updated the one occurence of underscore.

Also, it appeared that the dumping article was already rewritten during the opening of this PR and now. So I've removed that from the changes.

Thanks a lot and sorry for our slow responses on this PR.

**Comment by noniagriconomie** [comment]
@wouterj thank you

**Pull request closed without merging**